regression-tests.dnsdist/test_Lua.py

```diff
@@ -20,17 +20,19 @@ class TestLuaThread(DNSDistTest):
         counter = tonumber(data.i)
     end
 
-    newThread([==[
-      local socket = require'socket'
-      local i=1
-      while true
-      do
+    if inConfigCheck() == false then
```

Comment:
I'm not following (but this is likely lack of knowledge) how the thread is started. If the test fails, is this code re-executed at some later moment when the test is expected to succeed?

Also, what's the role of threadmessage?

Reply:
The testing framework (in dnsdisttests.py) runs `dnsdist --check-config` against the given config, before actually starting dnsdist. During check-config, we do not want this extra thread to generate messages, or run at all. `threadmessage` is called in the main Lua thread when this separate thread calls `submitToMainThread`.

Reply:
I looked up `inConfigCheck` and now its more clear. The question about `threadmessage` remains.

Reply:
Thanks. Searching for either `submitToMainThread` or `threadmessage` in the docs did not produce anything.

Reply:
This interface is (was?) likely to change before release, so we did not document it at the time. Now that it's in 1.7.0, I'll fix that.